WordPress/Classes/Utility/Analytics/WPAppAnalytics.m

```diff
@@ -0,0 +1,202 @@
+    }
+}
+
+#pragma mark - Notifications
+
+- (void)startObservingNotifications
+{
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(applicationDidBecomeActive:)
+                                                 name:UIApplicationDidBecomeActiveNotification
+                                               object:nil];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(applicationDidEnterBackground:)
+                                                 name:UIApplicationDidEnterBackgroundNotification
+                                               object:nil];
+}
+
+- (void)stopObservingNotifications
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+#pragma mark - Notifications
+
+- (void)applicationDidBecomeActive:(NSNotification*)notification
+{
+    [self trackApplicationOpened];
+}
+
+- (void)applicationDidEnterBackground:(NSNotification*)notification
+{
+    [self trackApplicationClosed];
+}
+
+#pragma mark - App Tracking
+
+/**
+ *  @brief      Tracks that the application has been closed.
+ */
+- (void)trackApplicationClosed
+{
+    NSMutableDictionary *analyticsProperties = [NSMutableDictionary new];
+    
+    analyticsProperties[WPAppAnalyticsKeyLastVisibleScreen] = self.lastVisibleScreenCallback();
+    
+    if (self.applicationOpenedTime != nil) {
+        NSDate *applicationClosedTime = [NSDate date];
+        NSTimeInterval timeInApp = round([applicationClosedTime timeIntervalSinceDate:self.applicationOpenedTime]);
+        analyticsProperties[WPAppAnalyticsKeyTimeInApp] = @(timeInApp);
+        self.applicationOpenedTime = nil;
+    }
+    
+    [WPAnalytics track:WPAnalyticsStatApplicationClosed withProperties:analyticsProperties];
+    [WPAnalytics endSession];
+}
+
+/**
+ *  @brief      Tracks that the application has been opened.
+ */
+- (void)trackApplicationOpened
+{
+    self.applicationOpenedTime = [NSDate date];
+    [WPAnalytics track:WPAnalyticsStatApplicationOpened];
+}
+
+#pragma mark - Usage tracking initialization
+
+- (void)initializeUsageTrackingIfNecessary
+{
+    if (![self isUsageTrackingInitialized]) {
+        [self setTrackingUsage:YES];
+        [NSUserDefaults resetStandardUserDefaults];
+    }
+}
+
+- (BOOL)isUsageTrackingInitialized
+{
+    return [[NSUserDefaults standardUserDefaults] valueForKey:WPAppAnalyticsDefaultsKeyUsageTracking] != nil;
+}
+
+#pragma mark - Usage tracking
+
+- (BOOL)isTrackingUsage
+{
+    return [[NSUserDefaults standardUserDefaults] boolForKey:WPAppAnalyticsDefaultsKeyUsageTracking];
+}
+
+- (void)setTrackingUsage:(BOOL)trackingUsage
+{
+    if (trackingUsage != [self isTrackingUsage]) {
+        [[NSUserDefaults standardUserDefaults] setBool:trackingUsage
+                                                forKey:WPAppAnalyticsDefaultsKeyUsageTracking];
+        
+        if (trackingUsage) {
```

Comment:
Yeah... I was actually wondering at some of the related code, but decided not to change the logic if possible.